single_kernel_mongo/managers/cluster.py

```diff
@@ -154,6 +157,17 @@ def update_config_server_db(self) -> None:
                 },
             )
 
+    def update_ldap_user_to_dn_mapping(self) -> None:
+        """Updates the ldap user to dn mapping value in the databag."""
+        self.assert_pass_hook_checks()
+        for relation in self.state.cluster_relations:
+            self.data_interface.update_relation_data(
```

Comment:
assert_pass_hook_checks doesn't check for leader and I think you will want a check for leader here  to avoid multiple events setting the relation databag

Reply:
We can add one more check but it's already checked in the caller.
Let me add it however indeed.